File: python/list2/centered_average.py
def centered_average(nums):

    #find the first occurences of the minimum and maximum values in the list
    minIndex = nums.index(min(nums))
    maxIndex = nums.index(max(nums))

    #make a new list without the maximum and minimum
    newList = [
        nums[i] for i in range(len(nums)) if i != minIndex and i != maxIndex
    ]

    #not necessary, but an extra check to make sure the length is not 0
    if len(newList) == 0: return 0

    #return the average
    return sum(newList) // len(newList)

File: java_code/array2/centeredAverage.java
package java_code.array2;

public class centeredAverage {

    public int centeredAverage(int[] nums) {
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;
        int sum = 0;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] <= min) {
                min = nums[i];
            }
            if (nums[i] >= max) {
                max = nums[i];
            }

            sum += nums[i];
        }

        return (sum - min - max) / (nums.length - 2);

    }

}